test/dsl001/test001.idr

```diff
@@ -35,8 +35,11 @@ using (G : Vect n Ty)
       If  : Expr G TyBool -> Expr G a -> Expr G a -> Expr G a
       Bind : Expr G a -> (interpTy a -> Expr G b) -> Expr G b
 
+  lam_ : TTName -> Expr (a :: G) t -> Expr G (TyFun a t)
+  lam_ _ = Lam
+
   dsl expr
-      lambda = Lam
```

Comment:
One thing I noticed, if I replace this with:
```
    lambda = (\_ => Lam)
```
or other lambda variants, idris ate up all available memory on my system while typechecking.
Not sure if this is known issue.